### PR TITLE
Amend Github action to build, test, push and deploy Docker image

### PR DESCRIPTION
**Pull request by StuartHarris**
# Amend Github action to build, test, push and deploy Docker image to `non-prod` cluster

Automates the deployment of services to the cluster(s) (in this PR, `non-prod` only).

Currently the Github action builds `doc-index-updater` and runs the tests. This PR moves the build and test into the Docker image build, pushes the image to the relevant container registry and updates the deployments repo with the new image digest (sha256). ArgoCD installed in the cluster sees this and deploys the new image.

### Acceptance Criteria

- [x] Docker image is built (and tests run)
- [x] Docker image is pushed to the relevant registry (master branch only)
- [x] `doc-index-updater` service is deployed to the cluster (on PR merge to master only for now)

### CODE REVIEW

File: medicines/doc-index-updater/examples/manifests/README.md [hunk 1]
```diff
@@ -28,20 +28,16 @@ curl -vvv http://doc-index-updater.localhost/non-existent-route # should be 404
...
 ### Encrypt the redis key as a Sealed Secret
 
-- retrieve the keys for the redis cache from the Azure Portal or the cli:
-
-```bash
-az redis list-keys --resource-group MHRA-dev --name doc-index-updater-dev --output tsv
-```
-
-- create the sealed secret with one of the above keys (replace `<insert key here>` with the key):
+- retrieve the keys for the redis cache from the Azure Portal or the cli, and create the sealed secret with one of the above keys:
```
Comment (rjkerrison) [inline medicines/doc-index-updater/examples/manifests/README.md]: doesn't make sense now, should say something like:

> create the sealed secret with keys for the redis cache retrieved from `az redis` cli,

File: medicines/doc-index-updater/examples/manifests/README.md [hunk 2]
```diff
@@ -50,52 +46,44 @@ kubectl create secret generic redis-key \
 
 #### Create Queue Key
 
-- retrieve the keys for the service bus access policy from the Azure Portal or the CLI:
+- retrieve the keys for the service bus access policy from the Azure Portal or the CLI, and create the sealed secret with one of the above keys:
```
Comment (rjkerrison) [inline medicines/doc-index-updater/examples/manifests/README.md]: again, should say that the command is 
> creating sealed secrets for the queue keys from `az servicebus`

File: medicines/doc-index-updater/examples/manifests/README.md [hunk 3]
```diff
@@ -50,52 +46,44 @@ kubectl create secret generic redis-key \
...
-  --resource-group MHRA-dev \
-  --namespace-name doc-index-updater-dev \
-  --queue-name doc-index-updater-delete-queue -\
-  -name doc-index-updater-delete-auth \
-  --query primaryKey \
-  --output tsv
-```
-
-- create the sealed secret with one of the above keys (replace `<insert key here>` with the key):
+- retrieve the keys for the service bus access policy from the Azure Portal or the CLI, and create the sealed secret with one of the above keys:
```
Comment (rjkerrison) [inline medicines/doc-index-updater/examples/manifests/README.md]: again, user of these docs just wants something that creates the keys, isn't interested in how they're retrieved necessarily

File: medicines/doc-index-updater/src/main.rs [hunk 4]
```diff
@@ -28,7 +28,8 @@ async fn main() -> Result<(), Box<dyn error::Error>> {
     let redis_addr = create_redis_url(redis_server, redis_port, redis_key);
 
     let time_to_wait = Duration::from_secs(get_env_or_default("SECONDS_TO_WAIT", 5));
-    let storage_container_name = get_env::<String>("STORAGE_CONTAINER").unwrap();
+    let storage_container_name =
+        get_env::<String>("STORAGE_CONTAINER").expect("STORAGE_CONTAINER env var not set");
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/main.rs]: 👍

File: medicines/doc-index-updater/examples/manifests/README.md (hunk 1 above)
Comment (StuartHarris) [reply medicines/doc-index-updater/examples/manifests/README.md]: I'm going to update this `manifests` example to point to the relevant manifests in the `deployments` repo. We really only want to maintain these in one place.

File: medicines/doc-index-updater/examples/manifests/README.md (hunk 2 above)
Comment (StuartHarris) [reply medicines/doc-index-updater/examples/manifests/README.md]: as above

File: medicines/doc-index-updater/examples/manifests/README.md (hunk 3 above)
Comment (StuartHarris) [reply medicines/doc-index-updater/examples/manifests/README.md]: as above

**Pull request merged**